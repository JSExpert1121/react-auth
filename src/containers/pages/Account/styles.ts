import { Theme, createStyles } from '@material-ui/core/styles';
import Colors from 'assets/colors';

export const styleFn = (theme: Theme) => createStyles({
    root: {
        width: '100%',
        padding: theme.spacing(3),
        maxWidth: 512,
        minWidth: 480,
    },
    form: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: '100%',
        textAlign: 'center'
    },
    logo: {
        padding: theme.spacing(2, 0)
    },
    title: {
        fontSize: '1.5rem',
        color: 'rgba(0, 0, 0, 0.8)',
        padding: theme.spacing(2)
    },
    text: {
        fontSize: '0.875rem',
        color: 'rgba(0, 0, 0, 0.7)',
        padding: theme.spacing(2)
    },
    success: {
        color: 'rgba(0, 0, 0, 0.9)',
        padding: theme.spacing(1)
    },
    error: {
        fontSize: '0.875rem',
        color: Colors.error,
        padding: theme.spacing(0, 0, 0, 2)
    },
    busy: {
        width: '100%',
    },
    submit: {
        margin: theme.spacing(2, 0),
        height: 'calc(100% - 12px)'
    },
    bottomMenu: {
        width: '100%',
        display: 'flex',
        padding: theme.spacing(2, 0)
    }
});