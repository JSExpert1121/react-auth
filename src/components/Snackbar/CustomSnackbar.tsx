import React from 'react';
import clsx from 'clsx';
import Fade from '@material-ui/core/Fade';
import IconButton from '@material-ui/core/IconButton';
import { Theme } from '@material-ui/core';
import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContent from '@material-ui/core/SnackbarContent';

import {
    CheckCircle as CheckCircleIcon,
    Close as CloseIcon,
    Error as ErrorIcon,
    Info as InfoIcon,
    Warning as WarningIcon,
} from '@material-ui/icons';

import { amber, green } from '@material-ui/core/colors';
import { makeStyles } from '@material-ui/styles';

const variantIcon = {
    success: CheckCircleIcon,
    warning: WarningIcon,
    error: ErrorIcon,
    info: InfoIcon,
};

const useStyles1 = makeStyles((theme: Theme) => ({
    success: {
        backgroundColor: green[600],
    },
    error: {
        backgroundColor: theme.palette.error.dark,
    },
    info: {
        backgroundColor: theme.palette.primary.dark,
    },
    warning: {
        backgroundColor: amber[700],
    },
    icon: {
        fontSize: 20,
    },
    iconVariant: {
        opacity: 0.9,
        marginRight: theme.spacing(1),
    },
    message: {
        display: 'flex',
        alignItems: 'center',
    },
}));

type SnackbarType = 'success' | 'warning' | 'error' | 'info';
interface ICustomSnackbarContentProps {
    className: string;
    message: string;
    variant: SnackbarType;
    onClose: () => void;
}

const CustomSnackbarContentWrapper: React.FC<ICustomSnackbarContentProps> = (props) => {
    const classes = useStyles1({});
    const { className, message, onClose, variant } = props;
    const Icon = variantIcon[variant];

    return (
        <SnackbarContent
            className={clsx(classes[variant], className)}
            aria-describedby="client-snackbar"
            message={
                <span id="client-snackbar" className={classes.message}>
                    <Icon className={clsx(classes.icon, classes.iconVariant)} />
                    {message}
                </span>
            }
            action={[
                <IconButton
                    key="close"
                    aria-label="Close"
                    color="inherit"
                    onClick={onClose}
                >
                    <CloseIcon className={classes.icon} />
                </IconButton>,
            ]}
        />
    );
}

const useStyles2 = makeStyles((theme: Theme) => ({
    margin: {
        margin: theme.spacing(1),
    }
}));

export interface ISnackbarProps {
    showMessage: boolean;
    message: string;
    variant: SnackbarType;
    handleClose: () => void;
}

interface ICustomSnackbarProps {
    open: boolean;
    message: string;
    variant: SnackbarType;
    handleClose: () => void;
}

const CustomSnackbar: React.FunctionComponent<ICustomSnackbarProps> = (props) => {
    const classes = useStyles2({});
    const { open, variant, message, handleClose } = props;
    return (
        <Snackbar
            anchorOrigin={{
                vertical: 'top',
                horizontal: 'center',
            }}
            open={open}
            autoHideDuration={4000}
            TransitionComponent={Fade}
            onClose={handleClose}
        >
            <CustomSnackbarContentWrapper
                onClose={handleClose}
                variant={variant}
                message={message}
                className={classes.margin}
            />
        </Snackbar>
    );
};

export default CustomSnackbar;
