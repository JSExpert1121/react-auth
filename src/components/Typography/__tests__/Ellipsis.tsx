import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import { createShallow, createMount } from '@material-ui/core/test-utils';

import EllipsisText from '../Ellipsis';

configure({ adapter: new Adapter() });

describe('components/Typography/Ellipsis', () => {

    const shallow = createShallow();
    const childTxt = 'Hello world';

    test('UI without', () => {
        const wrapper = shallow(<EllipsisText hidden={false}>{childTxt}</EllipsisText>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span')).toHaveLength(1);
        expect(wrapper.find('span').first().props().children).toEqual(childTxt);
        console.log(wrapper.find('span').first().props());
    });

    test('UI without', () => {
        const wrapper = shallow(<EllipsisText hidden={true}>{childTxt}</EllipsisText>);
        expect(wrapper).toMatchSnapshot();
    });
});