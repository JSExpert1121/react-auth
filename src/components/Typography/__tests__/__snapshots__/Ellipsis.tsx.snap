// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Typography/Ellipsis UI without 1`] = `ShallowWrapper {}`;

exports[`components/Typography/Ellipsis UI without 2`] = `ShallowWrapper {}`;
